src/wazuh_modules/syscollector/syscollector_bsd.c

```diff
@@ -63,410 +64,517 @@
 
-                            for (i = 0; _parts[i]; i++) {
-                                os_free(_parts[i]);
-                            }
-                            os_free(_parts);
+                        for (i = 0; _parts[i]; i++) {
+                            os_free(_parts[i]);
+                        }
+                        os_free(_parts);
 
-                            for (i = 0; parts[i]; i++) {
-                                os_free(parts[i]);
-                            }
-                            os_free(parts);
+                        for (i = 0; parts[i]; i++) {
+                            os_free(parts[i]);
                         }
+                        os_free(parts);
                     }
                 }
-
-            } else {  // Invalid file
-                mtdebug1(WM_SYS_LOGTAG, "Unable to read package information from '%s' (invalid format)", filepath);
-                invalid = 1;
             }
+        }
+    } else {
+        mtwarn(WM_SYS_LOGTAG, "Unable to read file '%s'", filepath);
+    }
+
+    cJSON_AddStringToObject(package, "location", app_folder);
+
+    if (!read_name) {
+        char * folder_cpy = NULL;
+        char * program_name = NULL;
+        char * end = NULL;
+
+        os_strdup(app_folder, folder_cpy);
+
+        // Extract program name from the path
+        program_name = strrchr(folder_cpy, '/');
+        program_name++;
+        end = strchr(program_name, '.');
+        *end = '\0';
 
+        if (normalize_mac_package_name(program_name, &vendor_name, &package_name)) {
+            cJSON_AddStringToObject(package, "name", package_name);
+            os_free(package_name);
+            if (vendor_name) {
+                cJSON_AddStringToObject(package, "vendor", vendor_name);
+                os_free(vendor_name);
+            }
         } else {
-            mtwarn(WM_SYS_LOGTAG, "Unable to read file '%s'", filepath);
+            cJSON_AddStringToObject(package, "name", program_name);
         }
+        os_free(folder_cpy);
+    }
 
-        if (strstr(app_folder, "/Utilities") != NULL) {
-            cJSON_AddStringToObject(package, "source", "utilities");
-        } else {
-            cJSON_AddStringToObject(package, "source", "applications");
+    fclose(fp);
+    return object;
+
+}
+
+/**
+ * @brief Read macOS applications installed by homebrew.
+ *
+ * @param app_folder Path to the directory to scan.
+ * @param timestamp Timestamp of the scan to be attached to the events.
+ * @param random_id Id of the scan, to be attached to the events.
+ * @param queue_fd File descriptor of the queue to send events.
+ * @param LOCATION Identifier of the component to attach it to the events.
+ * @return 0 on success, 1 otherwise.
+ **/
+int sys_read_homebrew_apps(const char * app_folder, const char * timestamp, int random_id, int queue_fd, const char* LOCATION) {
+
+    DIR *dr;
+    char path[PATH_LENGTH];
+    struct dirent *de;
```

Comment:
Please, initialize all variables whenever possible. For the array, instead of `memset()`, a more efficient solution is:
` char path[PATH_LENGTH] = {0};`

Reply:
Done